pyfarm/agent/sysinfo/network.py

```diff
@@ -93,11 +93,21 @@ def hostname(trust_name_from_ips=True):
     """
     logger.debug("Attempting to discover the hostname")
 
+    ip_addresses = addresses()
+
+    local_fqdn_query = socket.getfqdn().lower()
+    _h, _a, ips_from_fqdn = socket.gethostbyname_ex(local_fqdn_query)
```

Comment:
That's actually the wrong error code....it should come from the socket module.  Anyway I looked around and it looks like socket.gaierror does not have specific enough errors to have anymore specific behavior than just logging it (so nevermind on the if statement).